### ISSUE TITLE
nct6775 not detected on z170 chipset?

### ISSUE THREAD

**Issue by andygodber** [groeck/nct6775#39]
Hi Guenter 
3.10.0-327.10.1.v7.x86_64 kernel on Centos 7 using  ASRock z170 extreme 4 motherboard. Has 
nct6791d SuperIO chip.
Dreaded "could not insert nct6775 no such device' from modprobe.
sensors-detects finds something:
Trying family `VIA/Winbond/Nuvoton/Fintek'...               Yes
Found unknown chip with ID 0xc803
    (logical device B has address 0x290, could be sensors)

Any thoughts?
TIA
Andy

**Comment by groeck** [comment]
Support for NCT6791D was added to the driver after v3.10. I guess Centos/Redhat didn't back-port the updated driver. Did you try building and installing the driver from here ?

**Comment by pjperry** [comment]
elrepo.org have built this driver as a kmod package for RHEL/CentOS (kmod-nct6775) which you could try:

http://elrepo.org/linux/elrepo/el7/x86_64/RPMS/kmod-nct6775-0.0-3.20151222git.el7.elrepo.x86_64.rpm

**Comment by andygodber** [comment]
Thanks both - despite having thought I'd built from the GIT here, I must have messed up somewhere. So I retraced all my steps, deleted various copies of nct6775.ko from different location, and rebuilt from here - all worked ok.

Phil - I tried booting from the kernel (centos.plus) the repo from your suggestion created. Unfortunately this didnt work, leaving the same message 'no such device' from modprobe. Anything I can do to try and fix/help that?

Will close this thread later.
